src/lib/y2storage/guided_proposal.rb

```diff
@@ -57,11 +57,66 @@ class GuidedProposal
     # @return [Devicegraph]
     attr_reader :devices
 
+    class << self
+      # Calculates the initial proposal
+      #
+      # If a proposal is not possible by honoring current settings, other settings
+      # are tried. For example, a proposal without separate home or without snapshots
+      # will be calculated.
+      #
+      # @see GuidedProposal#initialize
+      #
+      # @param settings [ProposalSettings]
```

Comment:
And thinking about that (although I know it's not 100% related to this change)... It would be nice to improve the initialization to default values, so when `probed` is used we don't create a new analyzer but we also use `probed_disk_analyzer`. So just calling this with no arguments (so falling back to defaults) is more efficient. Ask me offline if my explanation is not understandable.